cogs/menudocs.py

```diff
@@ -97,6 +88,62 @@ async def on_thread_join(self, thread) -> None:
 
         await thread.join()
 
+    @commands.Cog.listener()
+    async def on_member_update(self, before: disnake.Member, after: disnake.Member):
+        if after.guild.id != MAIN_GUILD:
+            # Not in menudocs
+            return
+
+        if before.roles == after.roles:
+            # If roles are the same
+            return
+
+        welcome_messages = [
+            "$MEMBER just showed up. Hold my beer.",
+            "Where's $MEMBER? In the server!",
+            "Welcome, $MEMBER. Stay awhile and listen.",
+            "We've been expecting you, $MEMBER.",
+            "$MEMBER just joined. Can I get a heal?",
+            "$MEMBER just joined the server - glhf!",
+            "Ermagherd. $MEMBER has joined us!",
+            "Mission Control! $MEMBER has successfully landed!"
+        ]
+
+        projections = after.guild.get_channel(MENUDOCS_PROJECTIONS_CHANNEL)
+        unverified_role = after.guild.get_role(MENUDOCS_UNVERIFIED_ROLE)
+        message = Template(random.choice(welcome_messages)).safe_substitute(
+            {"MEMBER": f"**{after}**"}
+        )
+        embed = disnake.Embed(description=message, colour=disnake.Colour.green())
+
+        if unverified_role in before.roles:
```

Comment:
This also needs to check that the unverified role isn't in the after.roles, otherwise anything could trigger it